### PR TITLE
no-issue: Bump GHA versions to latest

### PR DESCRIPTION
**Pull request by rohan-bes**
We were getting warnings to upgrade these as node 16 is being deprecated

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -182,10 +182,10 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: 20.11.1
+          node-version-file: '.nvmrc'
           cache: yarn
       - run: yarn set version 3.2.1
```
Comment (jaskfla) [inline .github/workflows/ci.yaml:190]: Does the `actions/setup-node` installation of Node come with Corepack?

If so, we might be able to replace this with `corepack enable yarn` and have it read the Yarn version from `package.json` 🤔

(Disclaimer: I’m not saying this from experience; it just seems like something that _should_ work 😅)

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
     # Only run if Tupaia CI passed
     if: ${{ github.event.workflow_run.conclusion == 'success' }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v2
```
Comment (jaskfla) [inline .github/workflows/cd.yaml:21]: Possibly out of scope, but this is up to v4 now too. ([Changelog](https://github.com/aws-actions/configure-aws-credentials/blob/main/CHANGELOG.md))

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -17,26 +17,26 @@ jobs:
...
```
Comment (jaskfla) [inline .github/workflows/ci.yaml:41]: Does the `actions/setup-node` installation of Node come with Corepack?[^footnote]

[^footnote]: I couldn’t find any docs in their repo, but from what I understand Corepack’s been included with Node for quite a few major versions now—except if installing with Homebrew

If so, we might be able to replace these with `corepack enable yarn` and have it read the Yarn version from `package.json` 🤔

(Disclaimer: I’m not saying this from experience; it just seems like something that _should_ work 😅)

**Review by jaskfla — APPROVED** [review]
I’ve been thinking about doing this for a few weeks too now—thanks! 🙏

Great to see that the checks have passed 💃

Left a couple comments, but neither are strictly in scope for suppressing the Node  16 deprecation warnings—hence preapproving

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (rohan-bes) [reply .github/workflows/cd.yaml:21]: Good thinking, might as well try this upgrade. I'll spin up a deployment first to see if CD passes on it

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (rohan-bes) [reply .github/workflows/ci.yaml]: Seems to work! Thanks for the suggestion :)

**Pull request merged**